django/db/models/query.py

```diff
@@ -310,7 +310,7 @@ def iterator(self, chunk_size=2000):
         if chunk_size <= 0:
             raise ValueError('Chunk size must be strictly positive.')
         use_chunked_fetch = not connections[self.db].settings_dict.get('DISABLE_SERVER_SIDE_CURSORS')
-        return iter(self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size))
+        yield from self._iterable_class(self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size)
```

Comment:
This change is needed to pass `datetimes.tests.DateTimesTests.test_datetimes_has_lazy_iterator`.